atomix/cluster/src/main/java/io/atomix/raft/roles/FollowerRole.java

```diff
@@ -273,25 +255,6 @@ private void schedulePollRequests(final Duration delay) {
     }
   }
 
-  private void onHeartbeatTimeout(final Duration delay) {
-    if (!isRunning()) {
-      return;
-    }
-
-    final Duration pollTimeout = raft.getElectionTimeout();
-
-    heartbeatTimer =
-        raft.getThreadContext()
-            .schedule(
-                pollTimeout,
-                () -> {
-                  log.debug("Failed to poll a majority of the cluster in {}", pollTimeout);
-                  resetHeartbeatTimeout();
-                });
```

Comment:
Is it correct that we wait pollTimeout + rand ElectionTimeout again to trigger a new poll?

Reply:
Do you mean if we really want to wait for "pollTimeout + randElectionTimeout" again? I think it is debatable. A random element might be useful again to prevent two nodes from starting election at the same time.